.circleci/config.yml

```diff
@@ -152,13 +152,21 @@ jobs:
     executor: java-image
     steps:
       - checkout
-
       - restore_cache:
           keys:
             - v3-publish-dependencies-{{ checksum "pom.xml" }}
       - run: mvn versions:set -DnewVersion=${CIRCLE_TAG}
       - run: mvn deploy -Prun-revapi,public
 
+  dependency-track:
+    executor: java-image
+    steps:
+      - attach_workspace:
+          at: /root
+      - run:
+          name: Upload to Dependency Track
+          command: mvn verify -Pdependency-track
```

Comment:
Sorry I didn't notice it before... but in the pipeline, the tests fail on this step before dependency-track is run. However they are covered up with a build success because dependency-track works.

We could probably get away with `mvn compile -Pdependency-track`

Reply:
We don't see any test failures in the pipeline. Can you be more specific?

Having said that, we get your suggestion about doing it at the `compile` phase so that the tests do not run as part of the `dependency-track` step. This will require a change in root POM where the plugin is configured.

Reply:
sorry, let me know if you plan to make that change and I will help you fast track it...

If not its not too big of a deal.... in the raw output of dependency track I found the following exceptions and warnings which concerned me.

```
1827 [main] WARN org.eclipse.jetty.util.thread.QueuedThreadPool  - qtp1269742898{STOPPING,8<=8<=14,i=0,q=4} Couldn't stop Thread[qtp1269742898-42,5,]
qtp1269742898{STOPPING,8<=8<=14,i=0,q=4} Couldn't stop Thread[qtp1269742898-35,5,]
```

```
[INFO] Running com.autonomic.tmc.auth.ClientCredentialsTokenSupplierTest
qtp1269742898{STOPPING,8<=8<=14,i=0,q=4} Couldn't stop Thread[qtp1269742898-42,5,]
1827 [main] WARN org.eclipse.jetty.util.thread.QueuedThreadPool  - qtp1269742898{STOPPING,8<=8<=14,i=0,q=4} Couldn't stop Thread[qtp1269742898-42,5,]
```

Reply:
@integerAndrew These warning messages are coming from Wiremock and can be suppressed based on this: https://github.com/tomakehurst/wiremock/issues/563. We will change log level for this package in test.